test-case-reporting/types/test-case-reporting.d.ts

```diff
@@ -16,58 +16,53 @@
 
 declare module 'test-case-reporting' {
   /**
-   * A standard logging interface.
-   */
-  export interface Logger {
-    debug(message: string, ...extraInfo: unknown[]): void;
-    warn(message: string, ...extraInfo: unknown[]): void;
-    error(message: string, ...extraInfo: unknown[]): void;
-    info(message: string, ...extraInfo: unknown[]): void;
-  }
-
-  /**
-   * Possible job types.
+   * Travis job types for which test results may be reported.
    */
   export type JobType = 'unit' | 'integration';
 
-  /**
-   * Possible test statuses
-   */
   export type TestStatus = 'pass' | 'fail' | 'skip' | 'error';
 
   /**
-   * One concrete Travis build
+   * A build on Travis.
    */
   export interface Build {
     username: string,
     commitHash: string,
     prNumber: number,
     startedAt: Date,
+
+    // The list of jobs we know are contained in the Build.
+    // When we create a Build, we fill this with its Jobs, but when
+    //   we get a build from a server, the jobs list may not be populated
+    //   if it is more performant to omit them.
+    jobs: Array<Job>,
   }
 
   /**
-   * One concrete job within a build
+   * A job within a Travis build
    */
   export interface Job {
-    build: Build,
     jobNumber: string,
     type: JobType,
     startedAt: Date,
+
+    // This list is treated similarly to the `jobs` array of the `Build` type.
+    // Read the comment on `jobs` for info.
+    testRuns: Array<TestRun>,
   }
 
   /**
-   * One abstract kind of test case
+   * A single kind of test case, one `it` or `test` block
    */
   export interface TestCase {
     name: string,
     createdAt: Date,
   }
 
   /**
-   * One concrete run, in a job, of some testcase
+   * An instance of a test being run, with results
```

Comment:
hyper-nit: period at the end of these :)

Reply:
Done.